File: c/collatz-conjecture/collatz_conjecture.c
#include "collatz_conjecture.h"

inline int steps(int n) {
  int c = 0; if (n < 1) return ERROR_VALUE;
  while (n != 1) { n = n & 1? 3*n + 1 : n >> 1, ++c; } return c; }

File: fortran/collatz-conjecture/collatz_conjecture.f90
MODULE collatz_conjecture; CONTAINS
  PURE INTEGER FUNCTION steps(n) RESULT(c); VALUE n

    c = merge(0, -1, n > 0); DO WHILE (n > 1)
      c = c + 1; n = merge(n/2, 3*n + 1, iand(n, 1) == 0); END DO; END; END
